### ISSUE TITLE
Feature request: Create a Moodle H5P question type plugin

### ISSUE THREAD

**Issue by nadavkav** [h5p/h5p-moodle-plugin#104]
As our teachers use Moodle quizzes a lot, we keep getting requests to integrate H5P interactions into quizzes. Currently, we create a few H5P activities in the same course unit/section of the course. Hide them, then embed them inside "description" type quiz questions. But the gradebook gets messed up with grades of H5P activities outside the scope of the quiz. Also, it is relatively acceptable for a few H5P activities but not a good solution for a more serious usage of H5P.

We are already considering a development of Moodle H5P question type, and so before we start design and development... I am putting this issue here to see what you and the community thinks and plan for the future.

**Comment by mgancarzdsi** [comment]
Hi Nadav, I'm thinking about the same issue, though our challenge is we also want to embed h5p activities inside SCORM packages. 

For your issue, don't the gradebook items get hidden when you hide the h5p activity, and thus excluded from the calculation (though I think there may be an option for it)? You could also create a seperate grade category for those items and not include as part of the overall calculation?

My own plan so far is to have a seperate from Moodle Wordpress site that exists just to host H5P activities for embedding anywhere. The downside is it's another system/dependency to maintain.

**Comment by mgancarzdsi** [comment]
Wait, actually re-reading your comment, I see you mean an actual question type that you can add to activities, and feeds the score back into the quiz calculation as an overall part of the quiz grade.

I think this could be useful, but also tricky as it would then need to understand all of the quiz scoring/attempts/other Moodle quiz logistic things, resetting itself appropriately and also working with the various deferred and immediate feedback options of a quiz.

**Comment by nadavkav** [comment]
Indeed. not a small project ;-)

(Some teachers need the grades and some don't and just use it for student practice)

**Comment by gus900** [comment]
We are just looking at adopting H5P within our site, but not being able to integrate H5P content into our regular Moodle quizzes badly restricts its use.  Is anything happening in this area?

**Comment by falcon-git** [comment]
From the H5P core teams side nothing is happening in this area, but others in the community might be working on it. We do plan to add a way to add H5Ps to any other content in Moodle through either short codes or a new button in the WYSIWYG text editor.

**Comment by gus900** [comment]
Okay, many thanks for the response.  If anyone feels like working on this get in touch and maybe we can help each other, I've possibly got access to some funds but definitely have restricted access to expertise!

**Comment by dirkcgrunwald** [comment]
We would also be interested in a moodle quiz question type. We had a student work on an extension to the Moodle "opaque" question (soap interface to remote question server) to turn it into a REST interface to remote question server (see https://github.com/academicsystems/Qengine and https://github.com/academicsystems/moodle-qtype_opaque ).

I'll ask him to estimate the effort to do an H5P plugin. It would be useful if there was a really simplified example of the PHP side of H5P some place.

**Comment by thomasmars** [comment]
Looking forward to hearing more about this.
The closest thing to an overview of the PHP side of H5P can be found at the following links: 

- https://h5p.org/creating-your-own-h5p-plugin
- https://h5p.org/technical-overview

**Comment by jm-andonegi** [comment]
It's great to have an integration to add HP5 in Moodle as an Activity. Thanks! :)
It would be great to integrate HP5 in Moodle as a [Question type](url), so it can be included in the Question Bank.

**Comment by otacke** [comment]
Closing. As Svein-Tore stated, this is not something that H5P Group will be working on, and there are no newer plans. If someone still wants to build this, then discussion should take place in appropriate channels, e. g. the forums of H5P/Moodle or H5P's Matrix channel.
